### PR TITLE
feat: batch endpoints for courses/instructors

### PR DESCRIPTION
**Pull request by ecxyzzy**
## Description

Add support for fetching courses/instructors in a bulk, given a set of IDs.

## Motivation and Context

We have been able to support batch queries for a long time but have never done so (mostly due to my poor planning). This should remove the need for consumers to query our API _N_ times for _N_ things, if they already know exactly what they want.

## How Has This Been Tested?

Tested locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code involves a change to the database schema.
- [ ] My code requires a change to the documentation.

### CODE REVIEW

File: apps/api/src/schema/courses.ts [hunk 1]
```diff
@@ -41,6 +41,10 @@ export const coursesPathSchema = z.object({
     .openapi({ param: { name: "id", in: "path" } }),
 });
 
+export const batchCoursesQuerySchema = z.object({
+  ids: z.string().transform((xs) => xs.split(",")),
```
Comment (andrew-wang0) [inline apps/api/src/schema/courses.ts:45]:
```suggestion
  ids: z
    .string({ message: "Parameter 'ids' is required" })
    .transform((xs) => xs.split(","))
    .openapi({ example: "COMPSCI161,COMPSCI162" }),
```

File: apps/api/src/schema/instructors.ts [hunk 2]
```diff
@@ -6,6 +6,10 @@ export const instructorsPathSchema = z.object({
     .openapi({ param: { name: "ucinetid", in: "path" } }),
 });
 
+export const batchInstructorsQuerySchema = z.object({
+  ucinetids: z.string().transform((xs) => xs.split(",")),
```
Comment (andrew-wang0) [inline apps/api/src/schema/instructors.ts:10]:
```suggestion
  ucinetids: z
    .string({ message: "Parameter 'ucinetids' is required" })
    .transform((xs) => xs.split(","))
    .openapi({ example: "mikes,klefstad" }),
```

**Review by andrew-wang0 — APPROVED** [review]
:ok_hand:

File: apps/api/src/schema/courses.ts (hunk 1 above)
Comment (ecxyzzy) [reply apps/api/src/schema/courses.ts:45]: Good call, thanks for catching this!

**Review by andrew-wang0 — APPROVED** [review]
:ok_hand:

**Pull request merged**